src/backend/distributed/commands/alter_table.c

```diff
@@ -1532,6 +1523,50 @@ CreateMaterializedViewDDLCommand(Oid matViewOid)
 }
 
 
+/*
+ * This function renames the sequences owned by identities on the target relation by using
+ * sourceTableName
+ */
+static void
+RenameIdentitiesOnTable(char *sourceTableName, char *targetName, Oid targetRelationId)
+{
+	Relation relation = relation_open(targetRelationId, AccessShareLock);
+	TupleDesc tupleDescriptor = RelationGetDescr(relation);
+	relation_close(relation, NoLock);
+
+	StringInfo renameCommand = makeStringInfo();
+
+	for (int attributeIndex = 0; attributeIndex < tupleDescriptor->natts;
+		 attributeIndex++)
+	{
+		Form_pg_attribute attributeForm = TupleDescAttr(tupleDescriptor, attributeIndex);
+		char *columnName = NameStr(attributeForm->attname);
+		bool missing_ok = false;
+
+		if (attributeForm->attidentity)
+		{
+			Oid identitySchemaOid = get_rel_namespace(attributeForm->attrelid);
+			Oid seqOid = getIdentitySequence(targetRelationId, attributeForm->attnum,
+											 missing_ok);
+			char *oldSequenceName = generate_qualified_relation_name(seqOid);
+
+			char *newSequenceName = ChooseRelationName(sourceTableName,
```

Comment:
This works when the user hasn't previously renamed the sequences of generated identities.
I wonder if there is an easy way to actually get the names of the sequences of the old table before dropping it and save them in a list. and iterate that list here?

Otherwise we can print a NOTICE saying that all the sequences coming from generated identities in this table are renamed to their default Postgres names ?

Reply:
> I wonder if there is an easy way to actually get the names of the sequences of the old table before dropping it and save them in a list. and iterate that list here?

THIS + marking the identity sequence as distributed inside this `RenameIdentitiesOnTable` function could maybe fix all the problems?
@gokhangulbiz @marcocitus

Reply:
agree